engine/src/main/java/org/terasology/persistence/internal/EntitySetDeltaRecorder.java

```diff
@@ -83,9 +92,12 @@ private EntityDelta getOrCreateEntityDeltaFor(EntityRef entity) {
         return entityDelta;
     }
 
-    public void onEntityDestroyed(long entityId) {
-        entityDeltas.remove(entityId);
-        destroyedEntities.add(entityId);
+    public void onEntityDestroyed(EntityRef entity) {
+        if (!entity.isPersistent()) {
```

Comment:
I am aware of the arguments for guard statements, I just feel they only really apply if you methods are reasonably large or you have multiple guard statements. In this particular situation I wouldn't use them. This is just a personal style thing though.